### PR TITLE
update rpc deprecation notice

### PR DESCRIPTION
**Pull request by lostmygithubaccount**
## Description & motivation

We will not meet the current target of deprecating the RPC server by the end of 2022, which is a few weeks away. This updates to a reasonable ETA.

## To-do before merge
N/A

## Prerelease docs
If this change is related to functionality in a prerelease version of dbt (delete if not applicable):
- [ ] I've added versioning components, as described in ["Versioning Docs"](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/versioningdocs.md)
- [ ] I've added a note to the prerelease version's [Migration Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/guides/migration/versions)

## Checklist
If you added new pages (delete if not applicable):
- [ ] The page has been added to `website/sidebars.js`
- [ ] The new page has a unique filename

If you removed existing pages (delete if not applicable):
- [ ] The page has been removed from `website/sidebars.js`
- [ ] An entry has been added to `_redirects`
- [ ] [Run link testing](https://github.com/dbt-labs/docs.getdbt.com#running-the-cypress-tests-locally) to update the links that point to the deleted page

### CODE REVIEW

File: website/docs/reference/commands/rpc.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ description: "Remote Procedure Call (rpc) dbt server compiles and runs queries,
 You can use the `dbt-rpc` plugin to run a Remote Procedure Call (rpc) dbt server. This server compiles and runs queries in the context of a dbt project. Additionally, the RPC server provides methods that enable you to list and terminate running processes. We recommend running an rpc server from a directory containing a dbt project. The server will compile the project into memory, then accept requests to operate against that project's dbt context.
 
 :::caution Deprecation
-**The dbt-rpc plugin will be fully deprecated by the end of 2022.**
+**The dbt-rpc plugin will be fully deprecated by the second half of 2023.**
 
 For now, dbt Labs actively maintains and uses `dbt-rpc` to enable interactive dbt development. Once we announce the next-generation dbt Server is available for general release, we will deprecate the legacy plugin and only fix critical issues for a period of six months. After six months, we will archive this repository for read-only use.
```
Comment (jtcohen6) [inline website/docs/reference/commands/rpc.md]: > Once we announce the next-generation dbt Server is available for general release, we will deprecate the legacy plugin and only fix critical issues for a period of six months. After six months, we will archive this repository for read-only use.

Similar to https://github.com/dbt-labs/dbt-rpc/pull/104/files#r1044671376 - let's update this text too

**Comment by lostmygithubaccount** [comment]
related: https://github.com/dbt-labs/dbt-rpc/pull/104

File: website/docs/reference/commands/rpc.md [hunk 2]
```diff
@@ -16,9 +16,9 @@ description: "Remote Procedure Call (rpc) dbt server compiles and runs queries,
...
 
 :::caution Deprecation
-**The dbt-rpc plugin will be fully deprecated by the end of 2022.**
+**The dbt-rpc plugin will be fully deprecated by the second half of 2023.**
 
-For now, dbt Labs actively maintains and uses `dbt-rpc` to enable interactive dbt development. Once we announce the next-generation dbt Server is available for general release, we will deprecate the legacy plugin and only fix critical issues for a period of six months. After six months, we will archive this repository for read-only use.
+dbt Labs is actively maintaining `dbt-rpc` up to dbt v1.4. Starting in v1.5, we intend to break `dbt-rpc` compatibility in favor of [the new dbt Server](https://github.com/dbt-labs/dbt-server). dbt Labs will perform critical maintenance only, until the last compatible version of dbt has reached end of official support (12 months after release; [see Core version policies](/docs/dbt-versions/core)).
```
Comment (mirnawong1) [inline website/docs/reference/commands/rpc.md]: can we clarify where is dbt performing critical maintenance? 

>dbt Labs will perform critical maintenance only, until the last compatible version of dbt has reached end of official support (12 months after release; [see Core version policies](/docs/dbt-versions/core)).

Comment (mirnawong1) [inline website/docs/reference/commands/rpc.md]:
```suggestion
dbt Labs is actively maintaining `dbt-rpc` up to dbt v1.4. Starting in v1.5, we intend to break `dbt-rpc` compatibility in favor of [the new dbt Server](https://github.com/dbt-labs/dbt-server). dbt Labs will perform critical maintenance only, until the last compatible version of dbt has reached the end of official support (12 months after release; [see Core version policies](/docs/dbt-versions/core)).
```

**Review by mirnawong1 — DISMISSED** [review]
hey @lostmygithubaccount , this looks great and have some very small suggestions!

**Review by mirnawong1 — APPROVED** [review]
thanks @lostmygithubaccount !

**Pull request merged**